test/services/AuditServiceSpec.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import base.SpecBase
+import models._
+import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{verify, when}
+import play.api.libs.json.Json
+import services.AuditTypes.{NavigateToMessages, UploadSuccess}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.{AuditConnector, AuditResult}
+import uk.gov.hmrc.play.audit.http.connector.AuditResult.{Disabled, Failure, Success}
+import uk.gov.hmrc.play.audit.model.{DataEvent, ExtendedDataEvent}
+import uk.gov.hmrc.play.audit.AuditExtensions
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext.global
+
+class AuditServiceSpec extends SpecBase {
+  private val mockAuditConnector = mock[AuditConnector]
+  private val auditService = new AuditService(mockAuditConnector, appConfig)(global)
+  private val auditFailure = Failure("Event sending failed")
+
+  private val enrolment = AuthKey.enrolment
+  private val eori = "GB150454489082"
+  private val dcPath = "/secure-message-frontend/cds-file-upload-service/messages"
+  private val contactDetails = ContactDetails("Joe Bloggs", "Bloggs Inc", "0123456", "joe@bloggs.com")
+  private val fileUploadCount = FileUploadCount(1)
+  private val fileUpload = FileUpload("fileRef1", Successful, "", "id")
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    mockSendEvent()
+  }
+
+  "AuditService" should {
+
+    "audit a 'NavigateToMessages' event" in {
+      auditService.auditSecureMessageInbox(enrolment, eori, ExportMessages, dcPath)(hc)
+      verify(mockAuditConnector).sendExtendedEvent(ArgumentMatchers.refEq(navigateToMessageEvent, "eventId", "generatedAt"))(any(), any())
```

Comment:
I had to change all these tests using `ArgumentMatchers.refEq` because they were not actually working in Exports. To be sure, try replacing here `navigateToMessageEvent` as a quick test and check if you actually get an error.

Reply:
If I change a value in the `navigateToMessageEvent` I get the expected error:
`- should audit a 'NavigateToMessages' event *** FAILED ***
[info]   org.mockito.exceptions.verification.junit.ArgumentsAreDifferent: Argument(s) are different! Wanted:
[info] auditConnector.sendExtendedEvent(
[info]     refEq(ExtendedDataEvent(cds-file-upload-frontend,NavigateToMessages,b84fd9ba-e6ad-4578-93a0-96ed430b9493,Map(clientIP -> -, path -> /secure-message-frontend/cds-file-upload-service/messages, X-Session-ID -> -, Akamai-Reputation -> -, X-Request-ID -> -, deviceID -> -, clientPort -> -, transactionName -> callSFUSPartial),{"enrolment":"HMRC-CUS-ORG2","eoriNumber":"GB150454489082","tags":{"notificationType":"CDS-EXPORTS"}},2021-04-20T11:13:54.135Z)),
[info]     <any>,
[info]     <any>
[info] );
[info] -> at services.AuditServiceSpec.$anonfun$new$2(AuditServiceSpec.scala:56)
[info] Actual invocations have different arguments:
[info] auditConnector.sendExtendedEvent(
[info]     ExtendedDataEvent(cds-file-upload-frontend,NavigateToMessages,34b686bb-f16d-457b-a0dd-6c1caa2a4dec,Map(clientIP -> -, path -> /secure-message-frontend/cds-file-upload-service/messages, X-Session-ID -> -, Akamai-Reputation -> -, X-Request-ID -> -, deviceID -> -, clientPort -> -, transactionName -> callSFUSPartial),{"enrolment":"HMRC-CUS-ORG","eoriNumber":"GB150454489082","tags":{"notificationType":"CDS-EXPORTS"}},2021-04-20T11:13:54.208Z),
[info]     HeaderCarrier(None,None,None,None,RequestChain(51e9),13337986136601,List(),None,None,None,None,None,None,List()),`